src/read/elf/note.rs

```diff
@@ -87,6 +87,13 @@ where
     }
 }
 
+impl<'data, Elf: FileHeader> std::iter::Iterator for NoteIterator<'data, Elf> {
+    type Item = read::Result<Note<'data, Elf>>;
+    fn next(&mut self) -> std::prelude::v1::Option<Self::Item> {
+        self.next().transpose()
```

Comment:
This needs to check if an error occurred, and if it did then clear `self.data` so that the iterator is forced to stop at that point.